docs/rfcs/004-multi-file-diff.md

```diff
@@ -0,0 +1,65 @@
+# Multi-file Diffs
+
+## Status
+
+Proposed
+
+## Summary
+
+Users can select multiple files in unstaged changes and staged changes pane and see the diffs of selected files in one view, akin to the [`Files changed` tab in pull requests on github.com](https://github.com/atom/github/pull/1753/files).
+
+## Motivation
+
+So that users can view a full set of changes with more context before staging or committing those changes.
+
+The ability to display multiple diffs in one view will also serve as a building block for the following planned features:
+- [commit pane item](#1655) where it shows all changes in a single commit
+- [new PR review flow](https://github.com/atom/github/blob/master/docs/rfcs/003-pull-request-review.md) that shows all changed files proposed in a PR
+
+
+## Explanation
+
+#### Mock-ups coming soon
+
+
+#### Unstaged Changes pane
+- User can `cmd+click` and select multiple files from the list of unstaged changes, and the pane on the left (see multi-file diff section below) will show diffs of the selected files. That pane will continue to reflect any further selecting/unselecting on the Unstaged Changes pane.
+- Once there is at least one file selected, `Stage All` button should be worded as `Stage Selected`.
+
+#### Staged Changes pane
+- Same behavior as Unstaged Changes pane.
+- Once there is at least one file selected, `Unstage All` button should be worded as `Unstage Selected`.
+
+#### Multi-file diff view
+_(note: The following is a summary of what we would like the UX to achieve, but I don't have a clear visuals of what that looks like yet.)_
+
+- Shows diffs of multiple files as a stack.
+- Each diff should show up as its own block, and the current functionality should remain independent of each block.
```

Comment:
> Each diff should show up as its own block, and the current functionality should remain independent of each block.

Not quite sure what this means 🤔. Could you please elaborate?

Reply:
I realize now that it was quite badly worded. I meant the current functionality (e.g. the `open file`, `stage file`, etc from the header) of the diff view for the file it's showing should remain to be file-specific, even though the view would show more than one file. Does that clarify things? If not I can attempt to illustrate with mock-ups. Words are hardddd...

Reply:
I'm also not sure what this means.  In a visual sense, do you mean each file in the diff is separated visually by a border and some margin?

Reply:
Yup I think that makes sense! Thanks for the clarification. Maybe "Each diff retains the controls it currently has in its header (e.g. the open file, stage file, undo last discard, etc)"

Reply:
Oh interesting. So, effectively, we move the FilePatchItem's header controls as-is to the per-file banner within the multi file patch view? I like that - it'd be a lot easier than trying to figure out which file the file-specific buttons map to 😆